internal/jwtverify/token_verifier_jwt.go

```diff
@@ -58,6 +79,31 @@ type SubscribeTokenClaims struct {
 	jwt.StandardClaims
 }
 
+type jwksClient struct{ jwks.JWKSClient }
+
+func (j *jwksClient) verify(token *jwt.Token, kid string) error {
+	if kid == "" {
+		return errEmptyKeyID
+	}
+
+	key, err := j.JWKSClient.GetEncryptionKey(kid)
```

Comment:
I think we need a key with `"use": "sig"` here - i.e. for signature verification. Looking at go-jwks code I don't see how it differentiates "sig" and "enc" - both `GetEncryptionKey ` and `GetSignatureKey` return the same.

I suppose https://auth0.com/blog/navigating-rs256-and-jwks/ has a proper example of filtering keys received from endpoint:

```javascript
      const signingKeys = keys
        .filter(key => key.use === 'sig' // JWK property `use` determines the JWK is for signature verification
                    && key.kty === 'RSA' // We are only supporting RSA (RS256)
                    && key.kid           // The `kid` must be present to be useful for later
                    && ((key.x5c && key.x5c.length) || (key.n && key.e)) // Has useful public keys
        ).map(key => {
          return { kid: key.kid, nbf: key.nbf, publicKey: certToPEM(key.x5c[0]) };
        });
```

We need to do similar things, filter by `kty` (`RSA`), also do not use JWKS at all for unsupported algorithms (I think only `RS256` is commonly used at moment).

Reply:
Agree.